server/src/db/private_message.rs

```diff
@@ -0,0 +1,144 @@
+use super::*;
+use crate::schema::private_message;
+
+#[derive(Queryable, Identifiable, PartialEq, Debug, Serialize, Deserialize)]
+#[table_name = "private_message"]
+pub struct PrivateMessage {
+  pub id: i32,
+  pub creator_id: i32,
+  pub recipient_id: i32,
+  pub content: String,
+  pub deleted: bool,
+  pub read: bool,
+  pub published: chrono::NaiveDateTime,
+  pub updated: Option<chrono::NaiveDateTime>,
+}
+
+#[derive(Insertable, AsChangeset, Clone)]
+#[table_name = "private_message"]
+pub struct PrivateMessageForm {
+  pub creator_id: i32,
+  pub recipient_id: i32,
+  pub content: Option<String>,
+  pub deleted: Option<bool>,
+  pub read: Option<bool>,
```

Comment:
Why would deleted and read be options?

Reply:
A private message row can be "edited" by the creator to be deleted or have the content changed, and by the recipient to be read. They aren't required fields tho

Reply:
Oh right this is for the edit, nevermind then.